src/business/form.ts

```diff
@@ -17,44 +23,37 @@ export class Form {
             delete this.validationErrors[fieldName];
         });
 
-        this.triggerMultipleFieldListeners(fieldNames);
+        this.triggerMany(fieldNames);
     }
 
     public setFieldValue(fieldName: string, value: any) {
         this.values[fieldName] = value;
         delete this.validationErrors[fieldName];
 
-        this.triggerFieldListeners(fieldName);
+        this._fieldListenerRepository.trigger(fieldName, value);
     }
 
     public setValidationErrors(errors: { [fieldName: string]: string }) {
         this.validationErrors = errors;
 
         const fieldNames = Object.keys(this.validationErrors);
-        this.triggerMultipleFieldListeners(fieldNames);
+        this.triggerMany(fieldNames);
     }
 
     public getValidationError(fieldName: string): string | undefined {
         return this.validationErrors[fieldName];
     }
 
-    public listenForFieldChange(fieldName: string, callback: (value: any) => void) {
-        if (typeof this.fieldListeners[fieldName] === 'undefined') {
-            this.fieldListeners[fieldName] = [];
-        }
-        this.fieldListeners[fieldName].push(callback);
+    public subscribe(fieldName: string, callback: (value: any) => void): Listener {
+        return this._fieldListenerRepository.subscribe(fieldName, callback);
     }
 
-    private triggerMultipleFieldListeners(fieldNames: string[]) {
-        fieldNames.forEach((fieldName) => {
-            this.triggerFieldListeners(fieldName);
-        });
+    public unsubscribe(listener: Listener) {
+        this._fieldListenerRepository.unsubscribe(listener);
     }
 
-    private triggerFieldListeners(fieldName: string) {
-        if (typeof this.fieldListeners[fieldName] !== 'undefined') {
-            this.fieldListeners[fieldName].forEach((callback) => callback(this.getFieldValue(fieldName)));
-        }
+    private triggerMany(fieldNames: string[]) {
```

Comment:
Maybe wanna move this to the repo? Seems like a method the repo would want ownership of, if in the future it's behavior gotta change. (like we had before)